### ISSUE TITLE
Ideas

### ISSUE THREAD

**Issue by PechenkaPM** [MCMrARM/mc-w10-version-launcher#20]
Ideas for improvement:
1. Make the launcher installer (.msi)
2. Make an application icon
3. Make a GUI with the best design, without minimalism

**Comment by ItzLightyHD** [comment]
These are the same ideas i have for the next update.
I hope the GUI can be similar to the actual Minecraft Java Edition Launcher.
I like this project <3

**Comment by BlusterySasha-SoulMine** [comment]
+

**Comment by bullfrogberry** [comment]
This is Great. it is already known that some versions fail to download or are just not listed. BUT is there away if I have a beta installed that is NOT on the list (such as the new RTX Ray tracing specific Beta that I can copy this and add it as an option?

**Comment by ItzLightyHD** [comment]
> This is Great. it is already known that some versions fail to download or are just not listed. BUT is there away if I have a beta installed that is NOT on the list (such as the new RTX Ray tracing specific Beta 1.15.8.0
> 
> that I can copy this and add it as an option?
> 
> http://tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/e333c276-51c2-4cad-89b8-e193f1c86467?P1=1587139586&P2=402&P3=2&P4=bxUoqwRn3ASWI2GUHAKpjZ6Nx%2brLrDbWNANswJcIjTyzfVCAkd1EFMeIwQVDhU%2berXnJs6awlat1JrwHuGkfZQ%3d%3d

Yes, you can add it manually with only tweaking the "versions.json" file, if you can.

**Comment by bullfrogberry** [comment]
> > This is Great. it is already known that some versions fail to download or are just not listed. BUT is there away if I have a beta installed that is NOT on the list (such as the new RTX Ray tracing specific Beta 1.15.8.0
> > that I can copy this and add it as an option?
> > http://tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/e333c276-51c2-4cad-89b8-e193f1c86467?P1=1587139586&P2=402&P3=2&P4=bxUoqwRn3ASWI2GUHAKpjZ6Nx%2brLrDbWNANswJcIjTyzfVCAkd1EFMeIwQVDhU%2berXnJs6awlat1JrwHuGkfZQ%3d%3d
> 
> Yes, you can add it manually with only tweaking the "versions.json" file, if you can.
I tried this already but was unsure how it relates.
How would I add it manually? I see this type of code,
["1.14.20.1","d807bbf1-c63a-4d67-a53a-cc7d78398ffc",0]
how does the URL fit in here? I dont see any connection between the URL and the code

**Comment by techy804** [comment]
I think it's the part with "e333c276"

**Comment by dktapps** [comment]
You're not supposed to edit versions.json by hand. It's just a cache of https://mrarm.io/r/w10-vdb.
If you want to add version IDs that aren't already in the list, make a PR to this repo so that everyone can benefit: https://github.com/MCMrARM/mc-w10-versiondb/blob/master/versions.txt
